Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/non_rigid_mesh_registration.h

```diff
@@ -0,0 +1,856 @@
+    idx++;
+  }
+  SparseMat Nr(XF.rows() * XF.cols(), X.rows());
+  Nr.setFromTriplets(edge_coefficients.begin(), edge_coefficients.end());
+
+  // build MX
+  edge_coefficients.clear();
+  edge_coefficients.reserve(XF.rows() * XF.cols() * 2);
+  idx = 0;
+  for (Index i = 0; i < XF.rows(); ++i) {
+    int a = XF(i, 0);
+    int b = XF(i, 1);
+    int c = XF(i, 2);
+    edge_coefficients.push_back(SparseTriplet(idx, a, 1));
+    edge_coefficients.push_back(SparseTriplet(idx, b, -1));
+    idx++;
+    edge_coefficients.push_back(SparseTriplet(idx, b, 1));
+    edge_coefficients.push_back(SparseTriplet(idx, c, -1));
+    idx++;
+    edge_coefficients.push_back(SparseTriplet(idx, c, 1));
+    edge_coefficients.push_back(SparseTriplet(idx, a, -1));
+    idx++;
+  }
+  SparseMat B(XF.rows() * XF.cols(), X.rows());
+  B.setFromTriplets(edge_coefficients.begin(), edge_coefficients.end());
+  Index edim = B.rows();
+
+  Vertices BX = B * X;// edges in order: b to a, c to b, a to c.
+  Vertices BX_original(BX);
+
+  std::vector<SparseTriplet> coefficients;
+  SparseMat A(Z.rows() + 2 * dim + 3 * edim, dim + dim);
+  for (Index i = 0; i < dim; ++i) {
+    coefficients.push_back(SparseTriplet(Z.rows() + i, i, 1));
+  }
+
+  insertSparseMatrix(B, coefficients, Z.rows() + dim, 0);
+  insertSparseMatrix(B, coefficients, Z.rows() + dim + edim, Z.rows());
+  insertSparseMatrix(B, coefficients, Z.rows() + dim + 2 * edim, 2 * Z.rows());
+
+  Matrix b(Z.rows() + 2 * dim + 3 * edim, 1);
+
+  std::vector<SparseTriplet> weight_coefficients; // system regularizer
+  SparseMat W(dim + dim, dim + dim);
+  for (Index i = dim; i < dim + dim; ++i) {
+    weight_coefficients.push_back(SparseTriplet(i, i, 1));
+  }
+  W.setFromTriplets(weight_coefficients.begin(), weight_coefficients.end());
+
+  weight_coefficients.clear();
+  SparseMat D(Z.rows() + 2 * dim + 3 * edim, Z.rows() + 2 * dim + 3 * edim);
+  for (Index i = 0; i < Z.rows(); ++i) {
+    weight_coefficients.push_back(SparseTriplet(i, i, w2)); // point to plane energy
+  }
+  for (Index i = Z.rows(); i < Z.rows() + dim; ++i) {
+    weight_coefficients.push_back(SparseTriplet(i, i, w1)); // point to point energy
+  }
+  for (Index i = Z.rows() + dim; i < Z.rows() + dim + 3 * edim; ++i) {
+    weight_coefficients.push_back(SparseTriplet(i, i, w3)); // arap energy
+  }
+  D.setFromTriplets(weight_coefficients.begin(), weight_coefficients.end());
+
+  // Setting very high weight for given correspondences.
+  for (Index i = 0; i < corr.rows(); ++i) {
+    D.coeffRef(Z.rows() + corr(i, 0), Z.rows() + corr(i, 0)) = 1e15f;
+    D.coeffRef(Z.rows() * 2 + corr(i, 0), Z.rows() * 2 + corr(i, 0)) = 1e15f;
+    D.coeffRef(Z.rows() * 3 + corr(i, 0), Z.rows() * 3 + corr(i, 0)) = 1e15f;
+  }
+
+  // Solver
+  Eigen::ConjugateGradient<SparseMat, Eigen::Lower | Eigen::Upper> cg;
+  //cg.setMaxIterations(1000);
+  //cg.setTolerance(1e-6);
+  Eigen::JacobiSVD<Eigen::Matrix<ScalarType, 3, 3>> svd;
```

Comment:
I see a deprecated warning in [I-c-338](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-6.0.1-Ic-338/Polygon_mesh_processing_Examples/TestReport_lrineau_Ubuntu-GCC_master_CXX20-Release.gz)